src/methods.jl

```diff
@@ -10,318 +19,453 @@ derivatives.
-        """
-        ($D, $fdmf)
-    end
+julia> FiniteDifferences.estimate_step(fdm, sin, 1.0)  # Computes step size and estimates the error.
+(0.0010632902144695163, 1.9577610541734626e-13)
+```
+"""
+function (m::FiniteDifferenceMethod)(
+    f::Function,
+    x::T;
+    factor::Int=1,
+    max_step=convert(T, 0.1)
+) where T<:AbstractFloat
+    # The automatic step size calculation fails if `m.q == 0`, so handle that edge case.
+    iszero(m.q) && return f(x)
+    h, _ = estimate_step(m, f, x, factor=factor, max_step=max_step)
+    return m(f, x, h)
+end
+# Handle arguments that are not floats. Assume that converting to float is desired.
+function (m::FiniteDifferenceMethod)(f::Function, x::T; kw_args...) where T<:Real
+    return m(f, float(x); kw_args...)
 end
 
 """
-    FiniteDifferences.Nonstandard(grid, q; kwargs...)
+    (m::FiniteDifferenceMethod)(
+        f::Function,
+        x::T,
+        h
+    ) where T<:AbstractFloat
+
+Estimate the derivative of `f` at `x` using the finite differencing method `m` and a given
+step size.
+
+# Arguments
+- `f::Function`: Function to estimate derivative of.
+- `x::T`: Input to estimate derivative at.
+- `h`: Step size.
 
-An finite differencing method which is constructed based on a user-defined grid. It is
-nonstandard in the sense that it represents neither forward, backward, nor central
-differencing.
-See [`FiniteDifferenceMethod`](@ref) for further details.
+# Returns
+- Estimate of the derivative.
+
+# Examples
+
+```julia-repl
+julia> fdm = central_fdm(5, 1)
+FiniteDifferenceMethod:
+  order of method:       5
+  order of derivative:   1
+  grid:                  [-2, -1, 0, 1, 2]
+  coefficients:          [0.08333333333333333, -0.6666666666666666, 0.0, 0.6666666666666666, -0.08333333333333333]
+
+julia> fdm(sin, 1, 1e-3)
+0.5403023058679624
+
+julia> fdm(sin, 1, 1e-3) - cos(1)  # Check the error.
+-1.7741363933510002e-13
+```
 """
-function Nonstandard(grid::AbstractVector{<:Real}, q::Integer; adapt=0, kwargs...)
-    p = length(grid)
-    _check_p_q(p, q)
-    coefs = _coefs(grid, p, q)
-    hist = History(; adapt=adapt, kwargs...)
-    return Nonstandard{typeof(grid), typeof(coefs)}(Int(p), Int(q), grid, coefs, hist)
+function (m::FiniteDifferenceMethod)(
+    f::Function,
+    x::T,
+    h
+) where T<:AbstractFloat
+    return sum(
+        i -> convert(T, m.coefs[i]) * f(T(x + h * m.grid[i])),
+        eachindex(m.grid)
+    ) / h^m.q
 end
+# Handle arguments that are not floats. Assume that converting to float is desired.
+(m::FiniteDifferenceMethod)(f::Function, x::T, h) where T<:Real = m(f, float(x), h)
 
-# Check the method and derivative orders for consistency
+# Check the method and derivative orders for consistency.
 function _check_p_q(p::Integer, q::Integer)
-    q >= 0 || throw(ArgumentError("order of derivative must be nonnegative"))
+    q >= 0 || throw(ArgumentError("order of derivative must be non-negative"))
```

Comment:
We can take this chance to swap to `DomainError`s if we wanted 
```suggestion
    q >= 0 || throw(DomainError(q, "order of derivative (`q`) must be non-negative"))
```